### PR TITLE
Upgraded Compose and other dependencies. Removed kotlin-android-extensions

### PR DESCRIPTION
**Pull request by gpeal**
Tech debt dependencies cleanup.

The biggest change is migrating from kotlin-android-extensions to view binding and kotlin-parcelize

### CODE REVIEW

File: mvrx-navigation/src/test/kotlin/com/airbnb/mvrx/navigation/MavericksExtensionsTest.kt [hunk 1]
```diff
@@ -76,33 +76,4 @@ class MavericksExtensionsTest {
                 }
         }
     }
-
```
Comment (gpeal) [inline mvrx-navigation/src/test/kotlin/com/airbnb/mvrx/navigation/MavericksExtensionsTest.kt:79]: @marukami I'm not sure why this stopped throwing the expected exception but things still seem to be working okay. Can you verify?

**Review by elihart — APPROVED** [review]
Thanks for the cleanup!

File: buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -12,7 +12,7 @@ plugins {
 
 dependencies {
     implementation(gradleApi())
-    implementation("com.android.tools.build:gradle:7.0.0-alpha08")
+    implementation("com.android.tools.build:gradle:7.0.0-beta01")
     implementation("org.jacoco:org.jacoco.core:0.8.5")
-    implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.30")
+    implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.32")
```
Comment (inktomi) [inline buildSrc/build.gradle.kts:17]: One of the Compose drawbacks is the compiler plugin. Really, more of a Kotlin Compiler Plugin drawback.. unstable apis, and Compose projects will be a little behind on Kotlin versions for the foreseeable future..

File: buildSrc/src/main/java/dependencies.kt [hunk 3]
```diff
@@ -2,28 +2,28 @@ object Versions {
...
-    const val gradlePlugin = "7.0.0-alpha14"
+    const val gradlePlugin = "7.0.0-beta01"
     const val kotlin = "1.4.32"
     const val minSdk = 16
     const val targetSdk = 29
 
     // Android libraries
-    const val activity = "1.2.0"
+    const val activity = "1.2.3"
     const val activityCompose = "1.3.0-alpha03"
```
Comment (inktomi) [inline buildSrc/src/main/java/dependencies.kt:12]: Might as well update this one too. `1.3.0-alpha07` is the latest, I think. 

```suggestion
    const val activityCompose = "1.3.0-alpha07"
```

File: mvrx-navigation/src/test/kotlin/com/airbnb/mvrx/navigation/MavericksExtensionsTest.kt (hunk 1 above)
Comment (marukami) [reply mvrx-navigation/src/test/kotlin/com/airbnb/mvrx/navigation/MavericksExtensionsTest.kt:79]: recreate seems to wait for the resume state. Yer, seems to be fine. I'm not sure how to test the exception been thrown.

**Pull request merged**